documentation/src/main/asciidoc/topics/con_file_based_cache_stores.adoc

```diff
@@ -0,0 +1,77 @@
+[id='file-stores_{context}']
+= File-Based Cache Stores
+
+File-based cache stores provide persistent storage on the local host filesystem where {brandname} is running.
+For clustered caches, file-based cache stores are unique to each {brandname} node.
+
+You should use file-based cache stores to persist data that overflows from memory when the data container exceeds size or time restrictions.
+
+[WARNING]
+====
+Never use filesystem-based cache stores on shared file systems, such as an NFS or Samba share, because they do not provide file locking capabilities and data corruption can occur.
+
+Additionally if you attempt to use transactional caches with shared file systems, unrecoverable failures can happen when writing to files during the commit phase.
+====
+
+[discrete]
+== Soft-Index File Stores
+
+[NOTE]
+====
+`SoftIndexFileStore` is the default implementation for file-based cache stores.
+====
+
+`SoftIndexFileStore` is a Java implementation that uses a variant of **B+
+Tree** that is cached in-memory using Java soft references.
+The **B+ Tree**, called `Index` is offloaded on the file system to a single file that is purged and rebuilt each time the cache store restarts.
+
+`SoftIndexFileStore` stores data in a set of files rather than a single file.
+When usage of any file drops below 50%, the entries in the file are overwritten to another file and the file is then deleted.
+
+`SoftIndexFileStore` persists data in a set of files that are written in an append-only method.
+For this reason, if you use `SoftIndexFileStore` on conventional magnetic disk, it does not need to seek when writing a burst of entries.
+
+Most structures in `SoftIndexFileStore` are bounded, so out-of-memory exceptions do not pose a risk.
+You can also configure limits for concurrently open files.
+
+By default the size of a node in the `Index` is limited to 4096 bytes.
+This size also limits the key length, more precisely the length of serialized keys, which means you cannot use keys longer than the size of the node, 15
+bytes.
+Additionally, key length is stored as "short", which limits key length to 32767 bytes.
+`SoftIndexFileStore` throws an exception if keys are longer after serialization occurs.
+
+`SoftIndexFileStore` cannot detect expired entries, which can lead to excessive usage of space on the file system .
+
+[NOTE]
+====
+The `AdvancedStore.purgeExpired()` method is not implemented in `SoftIndexFileStore`.
+====
+
+.Segmentation
+
+Soft-index file stores are segmented by default with a separate instance per segment, which results in multiple directories.
```

Comment:
This is no longer true. There is only one SoftIndex instance and the index knows which segment each key maps to so it knows if an entry is needed to be returned or not.

Also we need to mention that soft index now only runs in segmented mode. We don't need to mention this in the docs but the benefit of 4 bytes per entry is vastly overshadowed by rehash and iteration performance improvements.

Reply:
How about if we simply to "Soft-index file stores are always segmented."?

Reply:
Sure.